pkgs/development/r-modules/default.nix

```diff
@@ -30,7 +30,12 @@ let
       urls = mkUrls (args // { inherit name version; });
     };
     inherit doCheck requireX;
-    propagatedBuildInputs = depends;
+    propagatedBuildInputs = depends  ++ ( with pkgs;
+      # packages assumed very often by R package tools are added to buildInputs
+      (lib.optionals stdenv.isDarwin [ libiconv flock which ])
+      ++ [ pkgs.gsl liblapack gmp.dev gtk2.dev bzip2.dev pkgs.pkgconfig
+           lzma icu.dev pcre.dev zlib.dev curl.dev libpng libtiff libjpeg
+         ]);
```

Comment:
This will increase the closure size quite a bit for packages that don't need this dependencies.

I'm assuming this is an optimization for your time since it is hard to track down who needs what, but for including this in nixpkgs I think it would be smart to add it on per package basis. I'm no expert in R, but I would assume this would apply also to rPackages. But I understand it creates much more work.

Reply:
Yes. I thought about that. If you add these it means a  new R packages is more likely to *just work*. However, what you say about closures is true. A mitigating factor is that the target audience for R mainly wants a development environment with many packages. In that scenario you only pay the size price once. If you think certain packages should be removed from this list I would be willing to add them explicitly as needed. However I don't thing the list should be totally empty. Thoughts?

Reply:
> A mitigating factor is that the target audience for R mainly wants a development environment with many packages. In that scenario you only pay the size price once.

I don't believe that you can legitimately speak for "the target audience" unless you have polled them first in some significant way. As it is, we don't know other people want. We can only speak about what *we* want.

Personally, I want accurate dependencies. I don't want packages to depend on stuff that they actually don't depend on. Furthermore, this is a guiding principle we have applied all over Nixpkgs, not just for R. This change goes against that design goal.

Reply:
In the current branch there are packages already added to every R derivation, whether they are needed or not. The question is what is the criteria for adding those?  I am sure I added some that are reasonable, but perhaps some should be done per-package. What's the right decision process? Should it include likelihood/ease of adding a future R package to nixpkgs? If not, we could make all dependencies explicit. More accurate, smaller closure size. But more breakage, more effort now and more in the future. I honestly don't know the correct answer here.

Reply:
On the target audience, it might be the case that the nix/R user only runs batch jobs and need only a few R packages at a time. That would be a fair comment. On the other hand, I can confidently say that there are few if any data scientists using nix/R to build and maintain a dev environment that includes bioconductor packages. That's what prompted me to attempt to improve the situation. Even if this PR got merged, serious problems remain for source packages that update frequently. We don't need a poll to see how conda supports those users better right now.

Reply:
> In the current branch there are packages already added to every R derivation, whether they are needed or not.

Which ones are you referring to?

> On the target audience, it might be the case that the nix/R user only runs batch jobs and need only a few R packages at a time.

Nobody said that. I all said was that I don't like it when people try to justify technical decisions with the needs of some "silent majority" without any evidence that they have actual knowledge about the needs of that group. I'd rather you speak for what *you* need than about what some mystical "target audience" needs.